s3transfer/futures.py

```diff
@@ -367,7 +369,8 @@ def _run_callback(self, callback):
 class BoundedExecutor(object):
     EXECUTOR_CLS = futures.ThreadPoolExecutor
 
-    def __init__(self, max_size, max_num_threads, tag_semaphores=None):
+    def __init__(self, max_size, max_num_threads, tag_semaphores=None,
+                 underlying_executor_cls=None):
```

Comment:
I don't think the `underlying` adds anything, this is being passed as a init param so it suggests that it's going to be used by this class.  I'd prefer just `executor_cls` like it's done in `manager.py`.